pyth-sdk/src/price.rs

```diff
@@ -89,6 +90,94 @@ impl Price {
         self.div(quote)?.scale_to_exponent(result_expo)
     }
 
+    /// Get the valuation of a collateral position according to:
+    /// 1. the net amount deposited (across the protocol)
+    /// 2. the max amount depositable (across the protocol)
+    /// 3. the initial and final valuation discount rates
+    /// 
+    /// We use a linear interpolation between the the initial and final discount rates,
+    /// scaled by the proportion of max depositable amount that has been deposited.
+    /// This essentially assumes a linear liquidity cumulative density function,
+    /// which has been shown to be a reasonable assumption for many crypto tokens in literature.
+    /// 
+    /// Args
+    /// deposits: u64, quantity of token deposited in the protocol
+    /// max_deposits: u64, max quantity of token that can be deposited in the protocol
+    /// discount_initial: u64, initial discount rate at 0 deposits (units given by discount_precision)
+    /// discount_final: u64, final discount rate at max_deposits deposits (units given by discount_precision)
+    /// discount_precision: u64, the precision used for discounts
```

Comment:
I think this should be provided as an exponent for consistency with the other functions in the sdk

Reply:
not sure what you're referring to, discount_precision?